pkg/linters/components/compimpl/linter_test.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package compimpl
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+func TestValid(t *testing.T) {
+	runLinter(t, func(string) string {
+		return `package component
+		func Module() fxutil.Module { return nil }
+		type Requires struct{}
+		type Provides struct{}
+		func NewComponent(_ Requires) Provides {
+			return Provides{}
+		}`
+	})
+}
+
+func TestMissingRequires(t *testing.T) {
+	runLinter(t, func(folderName string) string {
+		return fmt.Sprintf(
+			`package component // want "%v"
+		func Module() fxutil.Module { return nil }		
+		type Provides struct{}
+		func NewComponent(_ Requires) Provides {
+			return 0
```

Comment:
Is it intended to have these test functions return 0 instead of `Provides{}`? I feel like it could distract from what's actually being tested here if it's not necessary.